server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -141,6 +172,9 @@ public MultiSearchRequest(StreamInput in) throws IOException {
         super(in);
         maxConcurrentSearchRequests = in.readVInt();
         int size = in.readVInt();
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
nit: I think `size` on line 174 is related to `SearchRequest` array after it. So may be we should put reading `recallGoal` before `size` or after reading all search requests.